crates/common/tedge_config/src/tedge_config_cli/tedge_config_dto.rs

```diff
@@ -227,3 +225,77 @@ pub struct ServiceTypeConfigDto {
     #[serde(rename = "type")]
     pub(crate) service_type: Option<String>,
 }
+
+#[cfg(test)]
+mod tests {
+    use core::panic;
+    use std::borrow::Cow;
+
+    use figment::providers::Format;
+    use serde::de::DeserializeOwned;
+
+    use super::*;
+
+    #[test]
+    fn example_values_can_be_deserialised() {
+        let ty = TEdgeConfigDto::ty();
+        let doku::TypeKind::Struct { fields, transparent: false } = ty.kind else { panic!("Expected struct but got {:?}", ty.kind) };
+        let doku::Fields::Named { fields } = fields else { panic!("Expected named fields but got {:?}", fields)};
+        for (key, ty) in struct_field_paths(None, &fields) {
+            verify_examples_for::<TEdgeConfigDto>(&key, ty)
+        }
+    }
+
+    fn verify_examples_for<Dto>(key: &str, ty: doku::Type)
+    where
+        Dto: Default + Serialize + DeserializeOwned,
+    {
+        for example in ty.example.iter().flat_map(|e| e.iter()) {
```

Comment:
Thanks for this exhaustive test of all the examples. I tried it and yes, indeed, this catches mistakes in the doc.